app/[locale]/student-activities/clubs/page.tsx

```diff
@@ -1,9 +1,97 @@
-import WorkInProgress from '~/components/work-in-progress';
+import Link from 'next/link';
+import Image from 'next/image';
 
-export default function Clubs({
+import Heading from '~/components/heading';
+import { Card, CardDescription } from '~/components/ui';
+import { db } from '~/server/db';
+
+interface Club {
+  alias?: string | null;
+  name: string;
+  logo: string;
+  thumbnail: string;
+  urlName: string;
+  category: string;
+}
+
+export default async function StudentActivities({
   params: { locale },
 }: {
   params: { locale: string };
 }) {
-  return <WorkInProgress locale={locale} />;
+  const clubs: Club[] = await db.query.clubs.findMany({
+    columns: {
+      alias: true,
+      name: true,
+      logo: true,
+      thumbnail: true,
+      urlName: true,
+      category: true,
+    },
+  });
+
+  const groupedClubs: Record<string, Club[]> = clubs.reduce(
+    (acc: Record<string, Club[]>, club: Club) => {
+      const category = club.category;
+      if (!acc[category]) {
+        acc[category] = [];
+      }
+      acc[category].push(club);
+      return acc;
+    },
+    {}
+  );
+
+  console.log(groupedClubs);
+
+  return (
+    <>
+      <Heading
+        className="container"
+        glyphDirection="dual"
+        heading="h1"
+        href="#clubs-heading"
+        text="Clubs"
```

Comment:
Please use `~/i18n/translations.ts` so that all of our text is localised. Example usage:
https://github.com/nitkkr-dev/nitkkr/blob/fbc3d72b3cc92682b1219be78b65a71b61911b72/app/%5Blocale%5D/notifications.tsx#L18